src/js/requestHandlers.js

```diff
@@ -230,6 +254,74 @@ sjrk.storyTelling.server.deleteStoryFromCouch = function (id, deleteStoryDataSou
     return promise;
 };
 
+sjrk.storyTelling.server.deleteStoryFiles = function (handlerComponent, storyContent) {
+    var filesToDelete = [];
+
+    fluid.each(storyContent, function (block) {
+        var blockFileName = "";
+
+        if (block.blockType === "image") {
+            blockFileName = block.imageUrl;
+        } else if (block.blockType === "audio" || block.blockType === "video") {
+            blockFileName = block.mediaUrl;
+        }
+
+        if (sjrk.storyTelling.server.isValidMediaFilename(blockFileName)) {
+            filesToDelete.push(blockFileName);
+        }
+    });
+
+    // remove duplicate entries so we don't try to delete already-deleted files
+    filesToDelete = filesToDelete.filter(function (fileName, index, self) {
+        return self.indexOf(fileName) === index;
+    });
+
+    fluid.each(filesToDelete, function (fileToDelete) {
+        handlerComponent.deleteSingleFileRecoverable(fileToDelete);
+    });
+};
+
+/*
+ * Verifies that a given file name follows the UUID format as laid out in
+ * RFC4122. A detailed description of the format can be found here:
+ * https://en.wikipedia.org/wiki/Universally_unique_identifier#Format
+ */
+sjrk.storyTelling.server.isValidMediaFilename = function (fileName) {
+    if (fileName && typeof fileName === "string" ) {
+        var validUuid = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\..*$/;
+
+        return validUuid.test(fileName);
+    } else {
+        return false;
+    }
+};
+
+sjrk.storyTelling.server.getServerPathForFile = function (fileName, directoryName) {
+    return "." + directoryName + path.sep + fileName;
+};
+
+sjrk.storyTelling.server.deleteSingleFileRecoverable = function (fileToDelete, deletedFilesRecoveryPath, uploadedFilesHandlerPath) {
+    var recoveryPath = sjrk.storyTelling.server.getServerPathForFile(fileToDelete, deletedFilesRecoveryPath);
+    var deletionPath = sjrk.storyTelling.server.getServerPathForFile(fileToDelete, uploadedFilesHandlerPath);
+
+    // move it to the recovery dir and make sure it was moved
+    try {
+        fs.renameSync(deletionPath, recoveryPath);
+        fs.accessSync(recoveryPath, fs.constants.W_OK | fs.constants.R_OK);
+        fluid.log("Moved file to recovery dir:", recoveryPath);
```

Comment:
It would be helpful to provide the info of the source file location `deletionPath` in the message.

When deleted files are moved to a recovery directory, there are no side information saved along with these files such as where they come from or which story they belonged to, etc. This makes me wonder what's the purpose of having deleted files archived? For future recovery? Or security concerns to trace back? Also, when more and more files are accumulated in the recovery folder, we should think about the archive and clean-up rules for this folder. This doesn't need to be addressed here but it's probably worthwhile to note it as a to-do.